pkg/controller/node/controller_utils.go

```diff
@@ -67,7 +67,7 @@ func cleanupOrphanedPods(pods []*api.Pod, nodeStore cache.Store, forcefulDeleteP
 
 // deletePods will delete all pods from master running on given node, and return true
 // if any pods were deleted.
-func deletePods(kubeClient clientset.Interface, recorder record.EventRecorder, nodeName string, daemonStore cache.StoreToDaemonSetLister) (bool, error) {
+func deletePods(kubeClient clientset.Interface, recorder record.EventRecorder, nodeName, nodeUID string, daemonStore cache.StoreToDaemonSetLister) (bool, error) {
```

Comment:
We have the potential for delete-create-delete race in the node controller because we don't check uid.  Can you open an issue against the node controller that we need to precondition pod deletion against Pod uid?  There may be one open, but if not it's something that may be causing real problems today.